src/device-state.coffee

```diff
@@ -584,17 +584,19 @@ module.exports = class DeviceState extends EventEmitter
 				@applyBlocker
 		.then =>
 			@usingInferStepsLock =>
-				Promise.join(
-					@getCurrentForComparison()
-					@getTarget({ initial, intermediate })
-					(currentState, targetState) =>
-						@deviceConfig.getRequiredSteps(currentState, targetState)
-						.then (deviceConfigSteps) =>
-							if !_.isEmpty(deviceConfigSteps)
-								return deviceConfigSteps
-							else
-								@applications.getRequiredSteps(currentState, targetState, intermediate)
-				)
+				@applications.getExtraStateForComparison()
+				.then (extraState) =>
+					Promise.join(
```

Comment:
Can you use a `Promise.all` to keep the indentation a bit more manageable please?